internal/service/pushbasedlogexport/resource.go

```diff
@@ -0,0 +1,174 @@
+	resp.Diagnostics.Append(req.State.Get(ctx, &tfState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	connV2 := r.Client.AtlasV2
+	projectID := tfState.ProjectID.ValueString()
+	logConfig, _, err := connV2.PushBasedLogExportApi.GetPushBasedLogConfiguration(ctx, projectID).Execute()
+	if err != nil {
+		resp.Diagnostics.AddError("error getting push-based log export configuration", err.Error())
+		return
+	}
+
+	newTFModel, diags := NewTFPushBasedLogExport(ctx, projectID, logConfig, &tfState.Timeouts)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newTFModel)...)
+}
+
+func (r *pushBasedLogExportRS) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var tfPlan TFPushBasedLogExportRSModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &tfPlan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	logExportConfigReq := NewPushBasedLogExportReq(&tfPlan)
+
+	connV2 := r.Client.AtlasV2
+	projectID := tfPlan.ProjectID.ValueString()
+	if _, err := connV2.PushBasedLogExportApi.UpdatePushBasedLogConfiguration(ctx, projectID, logExportConfigReq).Execute(); err != nil {
+		resp.Diagnostics.AddError("error during push-based log export configuration update", err.Error())
+		return
+	}
+
+	timeout, diags := tfPlan.Timeouts.Update(ctx, defaultTimeout)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	logExportConfigResp, err := WaitStateTransition(ctx, projectID, connV2.PushBasedLogExportApi,
+		retryTimeConfig(timeout, minTimeoutCreateUpdate))
+	if err != nil {
+		resp.Diagnostics.AddError("error during push-based log export configuration update", err.Error())
+		return
+	}
+
+	newTFModel, diags := NewTFPushBasedLogExport(ctx, projectID, logExportConfigResp, &tfPlan.Timeouts)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newTFModel)...)
+}
+
+func (r *pushBasedLogExportRS) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var tfState *TFPushBasedLogExportRSModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &tfState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	connV2 := r.Client.AtlasV2
+	projectID := tfState.ProjectID.ValueString()
+	if _, err := connV2.PushBasedLogExportApi.DeletePushBasedLogConfiguration(ctx, projectID).Execute(); err != nil {
+		resp.Diagnostics.AddError("error during push-based log export configuration delete", err.Error())
+		return
+	}
+
+	deleteTimeout, diags := tfState.Timeouts.Delete(ctx, defaultTimeout)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	if err := WaitResourceDelete(ctx, projectID, connV2.PushBasedLogExportApi, retryTimeConfig(deleteTimeout, minTimeoutDelete)); err != nil {
+		resp.Diagnostics.AddError("error during push-based log export configuration delete", err.Error())
+		return
```

Comment:
Is this error going to be user visible? If so, please start the sentence with a capital letter, and state:
"Error when deleting a push-based log export configuration".

Please check other errors that are user visible to have their descriptions start with a capital letter and end with a period in the end.

Reply:
updated